Gameplay/PlayerController.cpp

```diff
@@ -0,0 +1,162 @@
+#include "scriptingUtil/gameplaypch.h"
+#include "PlayerController.h"
+#include <components/ComponentTransform.h>
+#include <components/ComponentCamera.h>
+#include <modules/ModuleSceneManager.h>
+
+Hachiko::Scripting::PlayerController::PlayerController(GameObject* game_object)
+	: Script(game_object, "PlayerController")
+	, _movement_speed(0.0f)
+	, _dash_distance(0.0f)
+	, _dash_duration(0.0f)
+	, _is_dashing(false)
+	, _dash_progress(0.0f)
+	, _dash_start(math::float3::zero)
+	, _rotation_speed(0.0f)
+	, _is_falling(false)
+	, _original_y(0.0f)
+	, _speed_y(0.0f)
+	, _starting_position(math::float3::zero)
+{
+}
+
+void Hachiko::Scripting::PlayerController::OnAwake()
+{
+	_dash_distance = 1.0f;
+	_dash_duration = 0.10f;
+	_movement_speed = 10.0f;
+	_rotation_speed = 2.5f;
+
+	_original_y = game_object->GetTransform()->GetGlobalPosition().y;
+	_speed_y = 0.0f;
+	
+	_starting_position = math::float3(0.0f, 1.0f, 0.0f);
+}
+
+void Hachiko::Scripting::PlayerController::OnStart()
+{
+}
+
+void Hachiko::Scripting::PlayerController::OnUpdate()
+{
+	// Ignore the inputs if engine camera input is taken:
+	if (Input::GetMouseButton(Input::MouseButton::RIGHT))
+	{
+		return;
+	}
+
+	ComponentTransform* transform = game_object->GetTransform();
+	
+	math::float3 current_position = transform->GetGlobalPosition();
+	math::float3 current_front = transform->GetFront();
+	math::float3 current_right = transform->GetRight();
+
+	float delta_time = Time::DeltaTime();
+	float velocity = _movement_speed * delta_time;
+
+	GameObject* raycast_hit = SceneManagement::Raycast(
+		current_position - math::float3(0.0f, 1.0f, 0.0f), current_position);
+
+	if (raycast_hit == nullptr || raycast_hit->name == "Player") 
+	{
+		_is_falling = true;
+	}
+
+	if (raycast_hit != nullptr && raycast_hit->name == "Goal")
+	{
+		SceneManagement::SwitchScene("Assets/Scenes/win.scene");
```

Comment:
Could be good to have these paths in a separate file all toghether. Mostly if we are using the same one in many places. What do you think?

Reply:
Yeah we can totally do that, we don't even know if these scene names will be used in the future, right now they don't even exist on the assets as we shifted to the YAML based serialization. For now, to stay up to your wishes, I'm creating a file for these in our scripting layer project "Gameplay".

Reply:
Cool!